### DISCUSSION TITLE
Using BlendSearch with ray

### DISCUSSION THREAD

**Discussion by jw00000** [microsoft/FLAML#248] [Q&A]
I'm trying to try use ray during fit, and I'm not sure why the below trivial example fails. This same example succeeds if I omit the n_concurrent_trials=10 arg. Seems like without this arg the search is done with CFO sequentially (instead of BlendSearch). It's my understanding that BlendSearch can be used to search the space in parallel. Any ideas? 

```
from sklearn.datasets import *
from flaml import AutoML
import ray
ray.init()
X,y = make_regression(1000,10)

automl = AutoML()
x = automl.fit(X,y,time_budget=120, task='regression', n_concurrent_trials=10)
```
The error: 

```
lib/python3.7/site-packages/flaml/tune/space.py in complete_config(partial_config, space, flow2, disturb, lower, upper)
    423             if upper and lower:
    424                 up, low = upper[key], lower[key]
--> 425                 gauss_std = up - low or flow2.STEPSIZE
    426                 # allowed bound
    427                 up += flow2.STEPSIZE

TypeError: unsupported operand type(s) for -: 'list' and 'list'
```

**Comment by sonichi** [comment]
You are right about when CFO and BlendSearch are used by default. The error is due to a bug when tuning multiple learners together in parallel. I created an issue #249 for it.
